storage-proofs/post/src/fallback/compound.rs

```diff
@@ -63,21 +64,25 @@ impl<'a, Tree: 'static + MerkleTreeTrait>
             // 1. Inputs for verifying comm_r = H(comm_c || comm_r_last)
             inputs.push(sector.comm_r.into());
 
+            // avoid rehashing fixed inputs
+            let mut challenge_hasher = Sha256::new();
+            challenge_hasher.update(AsRef::<[u8]>::as_ref(&pub_inputs.randomness));
+            challenge_hasher.update(&u64::from(sector.id).to_le_bytes()[..]);
+
```

Comment:
What is this? Is there supposed to be some changed behavior? It would really make current and historical review of this 45-file changeset simpler if it were restricted to *only* changes associated with introduction of this versioning scheme. If this is a simple refactor, it would be much better to put it in a distinct PR.

Reply:
Fair, but this brings these parts of the code up to date with the 'new normal' way of how these are done.

Reply:
@dignifiedquire Thoughts here? I can back this out of this PR to make a new PR for just those updates, but it seemed clear enough to me on this pass.  Either way, it's pretty straight-forward.

Reply:
can you put it into its own commit, fine in this PR

Reply:
For review, yes, I'll push shortly.  When we squash/merge it makes little difference in that case :-)

Reply:
I was thinking we squash the rest before merge and merge regularly so we end up with two commits on master

Reply:
Ok, updated into 2 commits, although I much prefer the squash/merge to the 'regular' merge.

Reply:
next time again